mas 1.8.2

phatblat opened the pull request:
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

SMillerDev commented on Formula/mas.rb:
```diff
@@ -13,27 +13,18 @@ class Mas < Formula
-    system "carthage", "bootstrap", "--platform", "macOS", "Commandant"
+    # system "script/build"
+    system "script/version"
+    system "swift", "build", "--disable-sandbox", "--manifest-cache", "none", "--configuration", "release",
+      "--arch", "arm64", "--arch", "x86_64"
```
Doesn't this try an build a universal binary?

chris-araman replied:
It does, as it did before. We have future work to make this architecture-specific.

carlocab replied:
Would passing the right architecture to `--arch` here build something architecture specific? If so, I think we can just make this
```suggestion
    system "swift", "build", "--disable-sandbox", "--manifest-cache", "none", "--configuration", "release",
      "--arch", Hardware::CPU.arch
```

chris-araman replied:
Removing the `--arch` flags would build for the current machine's architecture, but would change the path where it is built. We have some additional work before we get there. Changes to our build, install, and packaging scripts are necessary.

carlocab replied:
Where would it get installed? Does it affect `contrib/completion`? If not, then it seems to me that dropping the `--arch` flags would change only one more line here.

carlocab replied:
Can we `inreplace` the path in your install script here, so that it doesn't affect your package script?

chris-araman replied:
Potentially, if you think it's worth the hassle. `mas` 1.8.1 was also a universal binary, so this isn't a new change. I'm planning work to simplify and improve this for a future release—possibly as soon as 1.8.3.

carlocab replied:
> Potentially, if you think it's worth the hassle.

Please do. We've been very explicit about not supporting universal binaries, and I'd rather avoid users jumping to the conclusion that because building them for one formula can easily be done, it must be doable for others. (If they ever stumble upon the fact that `mas` is actually a universal binary on their system.)

chris-araman replied:
Will do. I'd hoped we could get our fixes out before doing this work, but it sounds like we'll need to get it done now.

chris-araman replied:
https://github.com/mas-cli/mas/pull/365#commits-pushed-3e88d28

carlocab commented:
```
error: unknown option --manifest-cache; use --help to list available options
```

SMillerDev commented:
I'm guessing that means it actually depends on xcode 12.5

chris-araman commented:
> I'm guessing that means it actually depends on xcode 12.5

It depends on 12.0 or 12.2 as declared, depending on architecture. However, in testing, we found that this flag was necessary when building with Xcode 12.5 installed. We should change this script to only pass this flag if 12.5 or later is installed.

phatblat commented:
I've simplified the formula to call our updated `build` script which addresses the issues identified in this PR, see mas-cli/mas#365, or specifics:

- [`build`](https://github.com/mas-cli/mas/pull/365/files#diff-99b83631df533ea1d9c97396a043a4670464f523d0c1e60b0a1cab6dddd440f9R9) now defaults to the current host architecture
- [`install`](https://github.com/mas-cli/mas/pull/365/files#diff-99b83631df533ea1d9c97396a043a4670464f523d0c1e60b0a1cab6dddd440f9R9) now knows how to find the binary, depending on how it was built

carlocab reviewed (approved):
Thanks, @phatblat, @chris-araman.
